docs/rules/require-default-props.md

```diff
@@ -197,13 +197,21 @@ NotAComponent.propTypes = {
 
 ```js
 ...
-"react/require-default-props": [<enabled>, { forbidDefaultForRequired: <boolean>, ignoreFunctionalComponents: <boolean> }]
+"react/require-default-props": [<enabled>, {
+  "forbidDefaultForRequired": <boolean>,
+  "ignoreFunctionalComponents": <boolean>,
+  "assignDefaultToObjectForFc": <boolean>
+}]
 ...
 ```
 
-* `enabled`: for enabling the rule. 0=off, 1=warn, 2=error. Defaults to 0.
-* `forbidDefaultForRequired`: optional boolean to forbid prop default for a required prop. Defaults to false.
-* `ignoreFunctionalComponents`: optional boolean to ignore this rule for functional components. Defaults to false.
+- `enabled`: for enabling the rule. 0=off, 1=warn, 2=error. Defaults to 0.
+- `forbidDefaultForRequired`: optional boolean to forbid prop default for
+  a required prop. Defaults to false.
+- `ignoreFunctionalComponents`: optional boolean to ignore this rule for
+  functional components. Defaults to false.
+- `assignDefaultToObjectForFc`: optional booolean to assign default to object
+  instead of `defaultProps` for functional components. Defaults to false.
```

Comment:
```suggestion
- `forbidDefaultForRequired`: optional boolean to forbid prop default for a required prop. Defaults to `false`.
- `ignoreFunctionalComponents`: optional boolean to ignore this rule for functional components. Defaults to `false`.
- `assignDefaultToObjectForFc`: optional boolean to assign default to object instead of `defaultProps` for functional components. Defaults to `false`.
```